examples/rust_caster.rs

```diff
@@ -264,16 +264,43 @@ fn play_media(
     }
 }
 
+
+fn discover() -> Option<(String, u16)> {
+    use mdns_sd::{ServiceDaemon, ServiceEvent};
+
+    let mdns = ServiceDaemon::new().expect("Failed to create daemon");
+
+    const SERVICE_TYPE: &str = "_googlecast._tcp.local.";
+    let receiver = mdns.browse(SERVICE_TYPE).expect("Failed to browse");
+
+    while let Ok(event) = receiver.recv() {
+        match event {
+            ServiceEvent::ServiceResolved(info) => {
+                return Some((info.get_addresses().iter().next().unwrap().clone().to_string(), info.get_port()))
+            }
+            _ => {}
+        }
+    }
+    return None
+}
+
+
 fn main() {
     env_logger::init();
 
-    let args: Args = Docopt::new(USAGE)
```

Comment:
__suggestion:__ maybe we can make this part a little bit easier to read (and avoid mutating `args`). How about something like this?
```rust
    let args: Args = Docopt::new(USAGE)
        .and_then(|d| d.deserialize())
        .unwrap_or_else(|e| e.exit());

    let (address, port) = match args.flag_address {
        Some(address) => (address, args.flag_port),
        None => {
            println!("Cast Device address is not specified, trying to discover...");
            discover().unwrap_or_else(|| {
                println!("No Cast device discovered, please specify device address explicitly.");
                std::process::exit(1);
            })
        }
    };

    let cast_device = match CastDevice::connect_without_host_verification(address, port) {
        Ok(cast_device) => cast_device,
        Err(err) => panic!("Could not establish connection with Cast Device: {:?}", err),
    };
```